container-engine-lib/lib/backend_interface/objects/service/service_config.go

```diff
@@ -8,6 +8,7 @@ import (
 	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_interface/objects/service_directory"
 	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_interface/objects/service_user"
 	"github.com/kurtosis-tech/stacktrace"
+	v1 "k8s.io/api/core/v1"
```

Comment:
I think this object should be agnostic to k8s or Docker, you will have to create a new Kurtosis object to support this. You can do it in a next PR or you can put a TODO comment for refactoring it later.

Reply:
I understand that this is only valid for K8s, but adding the K8s library here starts coupling the component to that specific implementation